core/src/scene/importer.cpp

```diff
@@ -0,0 +1,251 @@
+            m_textureNames.insert(name);
+            if (names.size() > 1) {
+                texture.push_back(normalizePath(name, parentPath));
+            } else {
+                texture = normalizePath(name, parentPath);
+            }
+        }
+    }
+}
+
+void Importer::normalizeSceneTextures(Node& root, const std::string& parentPath) {
+
+    if (Node textures = root["textures"]) {
+        for (auto texture : textures) {
+            if (Node textureUrl = texture.second["url"]) {
+                setNormalizedTexture(textureUrl, {textureUrl.Scalar()}, parentPath);
+            }
+        }
+    }
+
+    if (Node styles = root["styles"]) {
+
+        for (auto styleNode : styles) {
+
+            auto style = styleNode.second;
+            //style->texture
+            if (Node texture = style["texture"]) {
+                if (texture.IsScalar()) {
+                    setNormalizedTexture(texture, {texture.Scalar()}, parentPath);
+                }
+            }
+
+            //style->material->texture
+            if (Node material = style["material"]) {
+                for (auto& prop : {"emission", "ambient", "diffuse", "specular", "normal"}) {
+                    if (auto propNode = material[prop]) {
+                        if (auto matTexture = material[propNode]["texture"]) {
+                            if (matTexture.IsScalar()) {
+                                setNormalizedTexture(matTexture, {matTexture.Scalar()}, parentPath);
+                            }
+                        }
+                    }
+                }
+            }
+
+            //style->shader->uniforms->texture
+            if (Node shaders = style["shaders"]) {
+                if (Node uniforms = shaders["uniforms"]) {
+                    for (auto uniformNode : uniforms) {
+                        StyleUniform styleUniform;
+                        auto uniformValue = uniformNode.second;
+                        if (SceneLoader::parseStyleUniforms(uniformValue, nullptr, styleUniform) &&
+                                styleUniform.type == "sampler2D") {
+                            if (styleUniform.value.is<UniformTextureArray>()) {
+                                setNormalizedTexture(uniformValue,
+                                             styleUniform.value.get<UniformTextureArray>().names,
+                                             parentPath);
+                            } else {
+                                auto name = styleUniform.value.get<std::string>();
+                                setNormalizedTexture(uniformValue, {name}, parentPath);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+bool Importer::loadScene(const std::string& path) {
+
+    auto scenePath = path;
+    auto sceneString = stringFromFile(setResourceRoot(scenePath.c_str()).c_str(), PathType::resource);
```

Comment:
Yes thats a good catch. Instead of calling it again, I can just call it once from `tangram.cpp` and pass just the resultant `scenePath` to the `Importer::loadScene`.